controllers/cloudstackfailuredomain_controller.go

```diff
@@ -148,6 +154,21 @@ func (r *CloudStackFailureDomainReconciliationRunner) GetAllMachinesInFailureDom
 	return ctrl.Result{}, nil
 }
 
+// RequeueIfClusterNotReady check cluster to see if there is any rolling update going on.
+func (r *CloudStackFailureDomainReconciliationRunner) RequeueIfClusterNotReady() (ctrl.Result, error) {
```

Comment:
Interesting that there are two ways to check if a cluster is ready. Both the `status.Ready` and the `status.ControlPlaneReadyCondition`?

Are the two set under different circumstances?

Are we sure we want to prevent deletion if the control plane is in a not ready state? For example, what if it's not ready because it's deleting?

I'm sure that's not an issue as you must have ran this and it was fine.


In general though, could you please add some background (in comments is fine) on why this is necessary and what the difference between `status.Ready==true` and  `conditions.ControlPlaneReadyConditions` are?

Reply:
```Are we sure we want to prevent deletion if the control plane is in a not ready state? For example, what if it's not ready because it's deleting?
```
If that happens, that will be temporary since once deleting is done, new VM will be created due to reconcile and control plane will be ready again.

Reply:
If cluster is being deleted, a boot strap (kind) cluster will be used to delete current cluster and cluster will be paused while being deleted.

Reply:
Thank you for adding that conditional. Will merge now.